src/components/Editor.tsx

```diff
@@ -1,15 +1,107 @@
-import React, { Component } from 'react';
+import React, {Component, createRef, KeyboardEvent} from 'react';
 
-import styles from './Editor.module.scss';
+import {ARROW_LEFT, ARROW_RIGHT, BACKSPACE, META, ARROW_UP, ARROW_DOWN} from '../key';
+import {ISegment, SegmentType} from '../entities/segment';
+import {IEditorStateService, Subscriber} from '../services/editor-state.service';
+import {InsertTextAction} from '../state/insert-text.action';
+import {IEditorState} from '../state/editor.state';
+
+
+export default class Editor extends Component<any, IEditorState> {
+    private editorStateService = new IEditorStateService();
+
+    private readonly stateChangeSubscriber: Subscriber;
+
+    private editableRegion = createRef<HTMLDivElement>();
+
+    private isHoldingMetaKey = false;
+
+    constructor(props: any) {
+        super(props);
+        this.state = this.editorStateService.getState();
+
+        this.stateChangeSubscriber = () => this.setState(this.editorStateService.getState());
+        this.editorStateService.onStateChange(this.stateChangeSubscriber);
+    }
+
+    public componentWillUnmount() {
+        this.editorStateService.unSubscribe(this.stateChangeSubscriber);
+    }
 
-export class Editor extends Component<any, any> {
     render() {
         return (
-            <div
-                className={styles.Editor}
-                contentEditable
-                suppressContentEditableWarning
-            />
+            <div className={'Editor'}>
+                <div ref={this.editableRegion}
+                     className={'editable-region'}
+                     contentEditable
+                     suppressContentEditableWarning={true}
+                     onKeyDown={this.handleOnKeyDown}
+                     onKeyUp={this.handleOnKeyUp}
```

Comment:
```suggestion
```
Not in this PR

Reply:
but without this part, we cannot integrate the insert-action with the editor

Reply:
Oh, this is need! Sorry!

Reply:
we can only remove the onKeyUp

Reply:
@yibolu Let's follow your suggestions!